test/toc_obj.spec.js

```diff
@@ -87,4 +83,38 @@ describe('tocObj', () => {
 
     result.length.should.eql(0);
   });
+
+  it('empty text', () => {
+    const input = '<h1></h1>';
+    const result = tocObj(input);
+
+    result[0].text.should.eql('');
+  });
+
+  describe('children element', () => {
+    it('children + text', () => {
+      const input = [
+        '<h1 id="foo"><a>#</a>bar</h1>',
+        '<h1 id="foo">bar<a>#</a></h1>',
+        '<h1 id="foo"><a>#</a>bar<a>#</a></h1>'
```

Comment:
```suggestion
        '<h1 id="foo"><a>#</a>bar<a>#</a></h1>',
        '<h1 id="foo"><a>#</a><span>bar</span><a>#</a></h1>'
```

Reply:
`<h1 id="foo"><a>#</a><span>bar</span><a>#</a></h1>` would result in `#bar#` though since `<span>bar</bar>` is an element.

Reply:
A workaround is `if (element.type === 'text' || element.type === 'span')`.

`<h1><a>#</a><span>bar</span><a>#</a></h1>` -> `bar`

But the workaround is not foolproof,

`<h1><a>#</a>hey<span>bar</span><a>#</a></h1>` -> `hey`
`<h1><a>#</a><span>bar</span>hey<a>#</a></h1>` -> `bar`

Since `for..of` doesn't follow order, the result can be erratic.

Reply:
The children tag could be anything: `<span>`, `<a>` even `<p>`.